cmd/deploy/wait.go

```diff
@@ -66,48 +71,56 @@ func (dw *Waiter) wait(ctx context.Context, opts *Options) error {
 	return nil
 }
 
-func (dw *Waiter) waitForResourcesToBeRunning(ctx context.Context, opts *Options) error {
-	ticker := time.NewTicker(5 * time.Second)
-	to := time.NewTicker(opts.Timeout)
+func (dw *Waiter) waitForResourcesToBeRunning(ctx context.Context, opts *Options, namespace string) error {
 	c, _, err := dw.K8sClientProvider.ProvideWithLogger(okteto.GetContext().Cfg, dw.K8sLogger)
 	if err != nil {
 		return err
 	}
 
-	for {
-		select {
-		case <-to.C:
-			return fmt.Errorf("'%s' deploy didn't finish after %s", opts.Manifest.Name, opts.Timeout.String())
-		case <-ticker.C:
-			ns := okteto.GetContext().Namespace
-			dList, err := pipeline.ListDeployments(ctx, opts.Manifest.Name, ns, c)
-			if err != nil {
-				return err
-			}
-			areAllRunning := true
-			for i := range dList {
-				d := &dList[i]
-				if !deployments.IsRunning(ctx, ns, d.Name, c) {
-					areAllRunning = false
+	// first list the pods of the pipeline and check if they are running
+	podList, err := pipeline.ListPods(ctx, namespace, opts.Manifest.Name, c)
```

Comment:
What if the pod is not created because it has an error and it can't be created? I think it would not get any pod not checking that pod and could probably skip that one

Reply:
Yeah, there are cases in which the pod is not even created. For example, if the requests are too high and the quota is exceeded, the pod is not created. So, in that case, we would consider the dev environment as ready when resources where not created. This is the main reason why we were using deployments and statefulsets to check if they have some running replica